crates/ruff_db/src/parsed.rs

```diff
@@ -32,6 +32,8 @@ pub fn parsed_module(db: &dyn Db, file: File) -> ParsedModule {
             .extension()
             .map_or(PySourceType::Python, PySourceType::from_extension),
         FilePath::Vendored(_) => PySourceType::Stub,
+        // TODO(dhruvmanila): Virtual paths can be Jupyter Notebooks as well
+        FilePath::SystemVirtual(_) => PySourceType::Python,
```

Comment:
I think adding an extension method for virtual path should be possible if we document that a virtual path must:

* Use the file name as the last component
* Is only allowed to use `/` as separator.

Reply:
Do you think it's ok to use the `std::path::Path::extension` method to do this or should we add our own implementation?

Reply:
Not sure. It would need some tests to show that it's Okay. The alternative is to split of the part after the last `/` and then search for the last `.`